src/components/SessionInfoBar.tsx

```diff
@@ -1,15 +1,35 @@
+import * as HeartbeatActions from "@src/actions/heartbeat";
 import * as SessionActions from "@src/actions/session";
 import { useAppDispatch, useAppSelector } from "@src/hooks/app";
-import * as React from "react";
-import { BoxArrowRight } from "react-bootstrap-icons";
-import { QuestionCircleFill } from "react-bootstrap-icons";
-import { Clipboard } from "react-bootstrap-icons";
+import React, { useEffect } from "react";
+import {
+  BoxArrowRight,
+  CircleFill,
+  Clipboard,
+  ExclamationCircleFill,
+  QuestionCircleFill,
+} from "react-bootstrap-icons";
 
 export function SessionInfoBar() {
-  const { url, project_name, project_docs, user } = useAppSelector(
-    store => store.session.serverInfo
+  const { heartbeat, url, project_name, project_docs, user } = useAppSelector(
+    store => {
+      return {
+        ...store.session.serverInfo,
+        heartbeat: store.heartbeat,
+      };
+    }
   );
   const dispatch = useAppDispatch();
+
+  const checkHeartbeat = async () => {
+    dispatch(HeartbeatActions.heartbeatRequest());
+    setTimeout(checkHeartbeat, 60000);
+  };
+
+  useEffect(() => {
+    checkHeartbeat();
+  }, []);
```

Comment:
Does this empty dependency array mean this effect will only fire once? I see the timeout above, but I'm surprised that's not somehow included in the effect to trigger a re-render.

Reply:
Correct, an empty dependency array will only fire once. If you exclude the parameter, it will run every time. And if you include objects, it will fire whenever they change.

The setTimeout does work properly and re-fire the saga and trigger a re-render every minute as expected. That is due to `useEffect` being a hook.